tests/trustyai/drift/utils.py

```diff
@@ -0,0 +1,244 @@
+import json
+import os
+import subprocess
+from typing import Any, Dict, List, Optional
+
+import requests
+from kubernetes.dynamic import DynamicClient
+from kubernetes.dynamic.exceptions import NotFoundError
+from ocp_resources.inference_service import InferenceService
+from ocp_resources.namespace import Namespace
+from ocp_resources.pod import Pod
+from ocp_resources.route import Route
+from ocp_resources.trustyai_service import TrustyAIService
+from simple_logger.logger import get_logger
+from timeout_sampler import TimeoutSampler
+
+from tenacity import retry, stop_after_attempt, wait_exponential, retry_if_exception_type
+
+from tests.trustyai.constants import TIMEOUT_5MIN
+
+LOGGER = get_logger(name=__name__)
+TIMEOUT_30SEC: int = 30
+
+
+class TrustyAIServiceRequestHandler:
+    """
+    Class to encapsulate the behaviors associated to the different TrustyAIService requests we make in the tests
+    TODO: It will be moved to a more general file when we start using it in new tests.
+    """
+
+    def __init__(self, token: str, service: TrustyAIService, client: DynamicClient):
+        self.token = token
+        self.service = service
+        self.headers = {"Authorization": f"Bearer {self.token}", "Content-Type": "application/json"}
+        self.service_route = Route(
+            client=client, namespace=service.namespace, name="trustyai-service", ensure_exists=True
+        )
+
+    def _send_request(
+        self,
+        endpoint: str,
+        method: str,
+        data: Optional[Dict[str, Any]] = None,
+        json: Optional[Dict[str, Any]] = None,
+    ) -> Any:
+        url = f"https://{self.service_route.host}{endpoint}"
+
+        if method not in ("GET", "POST"):
+            raise ValueError(f"Unsupported HTTP method: {method}")
+        if method == "GET":
+            return requests.get(url=url, headers=self.headers, verify=False)
+        elif method == "POST":
+            return requests.post(url=url, headers=self.headers, data=data, json=json, verify=False)
+
+    def get_model_metadata(self) -> Any:
+        return self._send_request(endpoint="/info", method="GET")
+
+
+def create_ocp_token(namespace: Namespace) -> str:
+    return subprocess.check_output(["oc", "create", "token", "test-user", "-n", namespace.name]).decode().strip()
+
+
+def send_inference_request(
```

Comment:
Could this reuse the Inference code @rnetser has begun writing? https://github.com/opendatahub-io/opendatahub-tests/pull/21/files#diff-cda8dd882d87878926713495595a86da783b045b6bcb27eaa31e8b2957783104R26
Or the other way around, it does not seem to be doing anything exclusive to Trusty